clickhouse-client/src/test/java/com/clickhouse/client/ClickHouseNodeTest.java

```diff
@@ -438,4 +440,34 @@ public void testToUri() {
                         .toUri().toString(),
                 "http://server1.dc1:8123/db1?async=false&auto_discovery=true#apj,r1s1");
     }
+
+    @Test(groups = { "unit" }, dataProvider = "testPropertyWithValueList_endpoints")
+    public void testPropertyWithValueList(String endpoints, int numOfNodes, String[] expectedBaseUris) {
+        ClickHouseNodes node = ClickHouseNodes.of(endpoints);
+        Assert.assertEquals(node.nodes.size(), numOfNodes, "Number of nodes does not match");
+
+        int i = 0;
+        for (ClickHouseNode n : node.nodes) {
+            Assert.assertEquals(n.config.getDatabase(), "my_db");
+            Assert.assertEquals(expectedBaseUris[i++], n.getBaseUri());
+            String customSettings = (String)n.config.getOption(ClickHouseClientOption.CUSTOM_SETTINGS);
+            String configSettings = (String) n.config.getOption(ClickHouseClientOption.CUSTOM_SETTINGS);
+
+            Arrays.asList(customSettings, configSettings).forEach((settings) -> {
+                Map<String, String> settingsMap = ClickHouseOption.toKeyValuePairs(settings);
+                Assert.assertEquals(settingsMap.get("param1"), "value1");
+                Assert.assertEquals(settingsMap.get("param2"), "value2");
+            });
+        }
+    }
+
+    @DataProvider(name = "testPropertyWithValueList_endpoints")
+    public static Object[][] endpoints() {
+        return new Object[][] {
+            { "http://server1:9090/my_db?custom_settings=param1=value1,param2=value2", 1, new String[]{"http://server1:9090/"} },
```

Comment:
Is this a valid URL?

Reply:
it is not exactly URL - it is client endpoints spec. It may have multiple hosts for example. 
Current example has custom setting with multiple key-value pairs (what would be URL encoded to make it Web URL, but we do not need to do this until we send request - so it is all fine)